contracts/rust/deployer/src/builder.rs

```diff
@@ -785,19 +786,29 @@ impl<P: Provider + WalletProvider> DeployerArgs<P> {
             )
         })?;
 
-        tracing::info!(
-            "Transferring ownership of {} from EOA to {}",
-            target_contract,
-            new_owner
-        );
-
-        // Use the existing transfer_ownership function from lib.rs
-        let receipt =
+        // RewardClaim uses AccessControl instead of Ownable, so we need to grant the admin role
+        // instead of transferring ownership
+        let receipt = if contract_type == Contract::RewardClaimProxy {
+            tracing::info!(
+                "Granting DEFAULT_ADMIN_ROLE for {} to {} (RewardClaim uses AccessControl, not \
+                 Ownable)",
+                target_contract,
+                new_owner
+            );
+            crate::grant_admin_role(&self.deployer, contract_type, contract_address, new_owner)
```

Comment:
when using the flag `--transferOwnershipFromEoa`, if the target contract is the `RewardClaim` contract, then , use the `grant_admin_role` function to transfer ownership.